### PR TITLE
[WIP] Abort if StartAt or Next [state] is not valid

### PR DESCRIPTION
**Pull request by kbrock**
I tried to run a workflow with an invalid StartAt
This caused a strange error

before
------

```
lib/floe/workflow.rb:76:in `step_nonblock': undefined method `run_nonblock!' for nil:NilClass (NoMethodError)

      current_state.run_nonblock!
                   ^^^^^^^^^^^^^^
	from lib/floe/workflow.rb:63:in `step'
	from lib/floe/workflow.rb:58:in `run!'
	from exe/floe:37:in `<top (required)>'
```

after
-----

```
{"Error"=>"States.Runtime", "Cause"=>"State \"FirstState\" does not exist"}
```

### CODE REVIEW

**Comment by agrare** [comment]
@kbrock this seems like something we should catch at load time, at least StartAt and static Next states (maybe even validate all choice state targets are valid states)

**Comment by kbrock** [comment]
keep the spec.
make this a payload parsing check

**Comment by kbrock** [comment]
TODO: rail at load time rather than catching this exception.

I still feel we need better runtime failures, but agree this error in particular needs to happen at load/validation time

**Comment by kbrock** [comment]
not sure how this ties into #123 

Currently, not stating all the states is fine for tests.
Not sure what others feel about specifying partial workflows

File: lib/floe/workflow.rb [hunk 1]
```diff
@@ -103,7 +103,21 @@ def current_state
     private
 
     def step_next
-      context.state = {"Name" => context.next_state, "Input" => context.output} if context.next_state
+      desired_state = context.next_state || context.state_name
+      if !@states_by_name.key?(desired_state)
+        context.state["Error"] = "States.Runtime"
```
[the remark this answers is not in the document]
Comment (Fryguy) [reply lib/floe/workflow.rb:108]: We had a task/issue somewhere to validate workflows before we run them and raise exceptions early.  I'm wondering if now is the time to add in that logic, and then have a separate Workflow#validate or Workflow#valid? method with this as the first thing we check for.  Thoughts?

**Comment by kbrock** [comment]
Going with validation instead: https://github.com/ManageIQ/floe/issues/130

**Comment by agrare** [comment]
Hey @kbrock let me know what you think about https://github.com/ManageIQ/floe/pull/136 and if that covers the concerns you had with this

**Pull request closed without merging**